modules/opencensus/src/main/java/com/spotify/ffwd/opencensus/OpenCensusPluginSink.java

```diff
@@ -0,0 +1,168 @@
+ * limitations under the License.
+ * -/-/-
+ */
+
+
+package com.spotify.ffwd.opencensus;
+
+import com.google.inject.Inject;
+import com.spotify.ffwd.model.Batch;
+import com.spotify.ffwd.model.Metric;
+import com.spotify.ffwd.output.FakeBatchablePluginSinkBase;
+import com.spotify.ffwd.output.PluginSink;
+import eu.toolchain.async.AsyncFramework;
+import eu.toolchain.async.AsyncFuture;
+import io.opencensus.exporter.stats.stackdriver.StackdriverStatsConfiguration;
+import io.opencensus.exporter.stats.stackdriver.StackdriverStatsExporter;
+import io.opencensus.stats.Aggregation.Sum;
+import io.opencensus.stats.Measure.MeasureLong;
+import io.opencensus.stats.Stats;
+import io.opencensus.stats.StatsRecorder;
+import io.opencensus.stats.View.Name;
+import io.opencensus.stats.View;
+import io.opencensus.tags.TagContext;
+import io.opencensus.tags.TagContextBuilder;
+import io.opencensus.tags.TagKey;
+import io.opencensus.tags.TagValue;
+import io.opencensus.tags.Tagger;
+import io.opencensus.tags.Tags;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/***
+ * PluginSink to export metrics into Stackdriver via OpenCensus API.
+ *
+ * Right now, this is specifically designed to export event counts for
+ * use in SLAs but could easily be extended to support more view types
+ * by allowing them to be specified in config.
+ *
+ * Resources can be specified using the environment variables
+ * OC_RESOURCE_TYPE and OC_RESOURCE_LABELS until the api becomes stable
+ * https://www.javadoc.io/doc/io.opencensus/opencensus-api/latest/io/opencensus/resource/
+package-summary.html
+ */
+public class OpenCensusPluginSink extends FakeBatchablePluginSinkBase implements PluginSink  {
+  private static final Logger log = LoggerFactory.getLogger(OpenCensusPluginSink.class);
+  @Inject
+  private AsyncFramework async;
+
+  private Map<String, MeasureLong> measures;
+
+  private static final Tagger tagger = Tags.getTagger();
+  private static final StatsRecorder statsRecorder = Stats.getStatsRecorder();
+
+  private Optional<String> gcpProject;
+
+  public void init() {
+    // Is this actually needed, not sure. Better safe than sorry!
+    measures = new ConcurrentHashMap();
+  }
+
+  public OpenCensusPluginSink(Optional<String> gcpProject) {
+    this.gcpProject = gcpProject;
+  }
+
+  public void sendMetric(Metric metric) {
+    try {
+      MeasureLong measure = measures.get(metric.getKey());
+      if (measure == null) {
+        // If we keep getting new metrics, this will keep growing forever. But as
+        // there doesn't seem to be a way to unregister a view, it's not obvious what
+        // the right thing to do is here.
+        measure = MeasureLong.create("Events", "Number of Events", "1");
```

Comment:
should the measure be incremented with the value that is within `metric` and not just by 1?

Reply:
This is the dimension or unit used, so it might be m/s or Hertz. For dimensionless quantities, 1 seems to be the right answer.

Reply:
👍 let's try it out.